### ISSUE TITLE
FalsePositive on Undefined variables after a foreach loop

### ISSUE THREAD

**Issue by verfriemelt-dot-org** [phpstan/phpstan-strict-rules#138]
hi,

i think this example might lead to a false positive: https://phpstan.org/r/d5a7a671-bf7b-4d88-8c1b-955208256e21

phpstan complained about the variable possibly not beeing defined, which is why i added the check infront.
after enabling strict rules, the same error comes back up, which is a false positive i think:

```
Variable $lastValue might not be defined.
```

thank you :)
